server/app/services/docker/service_creator.rb

```diff
@@ -90,7 +90,7 @@ def client
     def build_env(instance_number)
       env = grid_service.env.dup || []
       env << "KONTENA_SERVICE_ID=#{grid_service.id.to_s}"
-      env << "KONTENA_SERVICE_NAME=#{grid_service.name}"
+      env << "KONTENA_SERVICE_NAME=#{grid_service.qualified_name}"
```

Comment:
NB: I kinda thing we should keep the `io.kontena.service.name` and `KONTENA_SERVICE_NAME` as-is for consistency with the server `GridService#name` etc.

This would then require adding a new `KONTENA_SERVICE=stack/name` and updating the `kontena/lb` to use that instead of `KONTENA_SERVICE_NAME` when it exists.